File: easy/1967-number-of-strings-that-appear-as-substrings-in-word/solution.java
/**
 * 1967. Number of Strings That Appear as Substrings in Word
 * Time: O(n * |word| * |pattern|)
 * Space: O(1)
 */
class Solution {
    public int numOfStrings(String[] patterns, String word) {
        int count = 0;
        for (String p : patterns) {
            if (word.contains(p)) {
                count++;
            }
        }
        return count;
    }
}

File: easy/1967-number-of-strings-that-appear-as-substrings-in-word/solution.py
from typing import List


class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        """
        1967. Number of Strings That Appear as Substrings in Word
        Time: O(n * |word| * |pattern|)
        Space: O(1)
        """
        return sum(1 for p in patterns if p in word)
